jetty-server/src/main/java/org/eclipse/jetty/server/ForwardedRequestCustomizer.java

```diff
@@ -639,19 +639,23 @@ else if (_for == null)
         @SuppressWarnings("unused")
         public void handlePort(HttpField field)
         {
+            String port = getLeftMost(field.getValue());
+            if (StringUtil.isEmpty(port))
+                throw new IllegalArgumentException(field.getName() + " has empty value");
+
             if (!getForwardedPortAsAuthority())
             {
                 if (_for == null)
-                    _for = new PortSetHostPort(_request.getRemoteHost(), Integer.parseInt(getLeftMost(field.getValue())));
+                    _for = new PortSetHostPort(_request.getRemoteHost(), Integer.parseInt(port));
```

Comment:
Perhaps add a new method called `parsePort(String rawPort)` that does the Integer.parseInt() and also validates the range of values to be sane (`port >= 1 && port <= 65535`).